src/gui_common/CustomRichTextLabel.cs

```diff
@@ -0,0 +1,245 @@
+
+                // Is a closing tag
+                if (tagNamespace.BeginsWith("/"))
+                {
+                    // Closing tag doesn't match opening tag or vice versa, aborting parsing
+                    if (tagStack.Count == 0 || tagStack.Peek() != tagIdentifier)
+                    {
+                        result.Append($"[{tagBlock}]");
+                        isIteratingTag = false;
+                        continue;
+                    }
+
+                    // Finally try building the bbcode template for the tagged substring
+
+                    var closingTagStartIndex = ExtendedBbcode.Find("[", lastStartingTagEndIndex);
+
+                    var input = ExtendedBbcode.Substr(
+                        lastStartingTagEndIndex + 1, closingTagStartIndex - lastStartingTagEndIndex - 1);
+
+                    ThriveBbCodeTag parsedTag;
+
+                    if (Enum.TryParse(tagStack.Peek(), true, out parsedTag))
+                    {
+                        // Success!
+                        result.Append(BuildTemplateForTag(input, parsedTag));
+                    }
+                    else
+                    {
+                        // Tag is not present in the enum
+                        result.Append(input);
+                        GD.PrintErr($"Failed parsing custom thrive tag: {tagStack.Peek()}, it probably doesn't exist");
+                    }
+
+                    isIteratingContent = false;
+                    tagStack.Pop();
+                }
+                else
+                {
+                    isIteratingContent = true;
+                    tagStack.Push(tagIdentifier);
+                }
+
+                lastStartingTagEndIndex = index;
+
+                // Finished iterating tag
+                isIteratingTag = false;
+            }
+        }
+
+        // Apply the final string into this RichTextLabel's bbcode text
+        BbcodeText = result.ToString();
+    }
+
+    /// <summary>
+    ///   Returns a templated bbcode string for the given custom tag.
+    /// </summary>
+    /// <param name="input">The string tagged by custom tags</param>
+    /// <param name="tag">Custom Thrive bbcode-styled tags</param>
+    private string BuildTemplateForTag(string input, ThriveBbCodeTag tag)
+    {
+        // Defaults to input so if something fails output returns unchanged
+        var output = input;
+
+        switch (tag)
+        {
+            case ThriveBbCodeTag.Compound:
+            {
+                if (SimulationParameters.Instance.DoesCompoundExist(input))
+                {
+                    var compound = SimulationParameters.Instance.GetCompound(input);
+
+                    output = $"[b]{compound.Name}[/b] [font=res://src/gui_common/fonts/" +
+                        $"BBCode-Image-VerticalCenterAlign-3.tres] [img=20]{compound.IconPath}[/img][/font]";
+                }
```

Comment:
I think printing a warning if not found would be a good idea to make it more visible that there is some invalid bbcode.